addons/website/static/src/snippets/s_searchbar/000.js

```diff
@@ -0,0 +1,189 @@
+     */
+    _fetch: function () {
+        return this._rpc({
+            route: '/website/snippet/autocomplete',
+            params: {
+                'search_type': this.searchType,
+                'term': this.$input.val(),
+                'order': this.order,
+                'limit': this.limit,
+                'max_nb_chars': Math.round(Math.max(this.autocompleteMinWidth, parseInt(this.$el.width())) * 0.22),
+                'options': this.options,
+            },
+        });
+    },
+    /**
+     * @private
+     */
+    _render: function (res) {
+        var $prevMenu = this.$menu;
+        this.$el.toggleClass('dropdown show', !!res);
+        if (res) {
+            var results = res['results'];
+            let template = 'website.s_searchbar.autocomplete';
+            const candidate = template + '.' + this.searchType;
+            if (qweb.has_template(candidate)) {
+                template = candidate;
+            }
+            this.$menu = $(qweb.render(template, {
+                results: results,
+                parts: res['parts'],
+                hasMoreResults: results.length < res['results_count'],
+                widget: this,
+            }));
+            this.$menu.css('min-width', this.autocompleteMinWidth);
+            this.$el.append(this.$menu);
+            this.$el.find('button.extra_link').on('click', function (event) {
+                event.preventDefault();
+                window.location.href = event.currentTarget.dataset['target'];
+            });
+        }
+        if ($prevMenu) {
+            $prevMenu.remove();
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _onInput: function () {
+        if (!this.limit) {
+            return;
+        }
+        this._dp.add(this._fetch()).then(this._render.bind(this));
+    },
+    /**
+     * @private
+     */
+    _onFocusOut: function () {
+        if (!this.$el.has(document.activeElement).length) {
+            this._render();
+        }
+    },
+    /**
+     * @private
+     */
+    _onKeydown: function (ev) {
+        switch (ev.which) {
+            case $.ui.keyCode.ESCAPE:
+                this._render();
+                break;
+            case $.ui.keyCode.UP:
+            case $.ui.keyCode.DOWN:
+                ev.preventDefault();
+                if (this.$menu) {
+                    let $element = ev.which === $.ui.keyCode.UP ? this.$menu.children().last() : this.$menu.children().first();
+                    $element.focus();
+                }
+                break;
+        }
+    },
+    /**
+     * @private
+     */
+    _onSearch: function (ev) {
+        if (!this.$input[0].value) { // clear button clicked
+            this._render(); // remove existing suggestions
+            this.limit = 0; // prevent autocomplete
```

Comment:
This will prevent autocomplete when user is typing (after click on clear button).